src/CodeGen_Hexagon.cpp

```diff
@@ -2324,11 +2183,9 @@ bool CodeGen_Hexagon::possiblyCodeGenWideningMultiply(const Mul *op) {
     Ops.clear();
     //   Vdd.uh=vmpy(Vu.ub,Rt.ub)
     //   Vdd.uw=vmpy(Vu.uh,Rt.uh)
-    Patterns.push_back(Pattern(WPICK(wild_u16x128, wild_u16x64) *
-                               WPICK(wild_u16x128, wild_u16x64),
+    Patterns.push_back(Pattern(wild_u16x2W * wild_u16x2W,
```

Comment:
Yes, IIRC these are expected to be double vector wide. Taking Dillon's example 'a' is u16x32 with the upcast we get
u32x32 which is two vectors long. We want to detect double wide vectors and then see if the double vectors are themselves upcasts of narrower types. Makes sense?